lib/rspec/rails/matchers/be_a_new.rb

```diff
@@ -49,7 +49,7 @@ def attributes
 
         def attributes_match?(actual)
           attributes.stringify_keys.all? do |key, value|
-            actual.attributes[key].eql?(value)
+            actual.attributes[key].eql?(value) || (value.respond_to?(:expected, true) && actual.attributes[key].include?(value.expected))
```

Comment:
To work with all matchers, you need to delegate to the logic of each matcher.  The way we do that for composable matchers is via the `values_match?(expected, actual)` method:

https://github.com/rspec/rspec-expectations/blob/0ad53b7b0a6e34b1d20f64b1c6588a26648cf6ed/lib/rspec/matchers/composable.rb#L51-L69

That said, the `HaveAttributes` matcher already does this (and more) and given that the attribute matching here should have the same semantics, I think it might be best if we just delegate to that matcher here:

``` ruby
def attributes_match?(actual)
  HaveAttributes.new(attributes).matches?(actual)
end
```

If we go that route, we'll probably want to delegate to `HaveAttributes` for part of the failure message, too.

All that said: I don't know much about rails these days and I'm not sure if the `attributes` method here needs to be used.  @JonRowe or @samphippen -- can one of you advise?